app/webpacker/components/Results/Rankings/RankingsTable.jsx

```diff
@@ -18,15 +18,16 @@ function CountryCell({ country }) {
 }
 
 function ResultRow({
-  result, competition, rank, isAverage, show, country,
+  result, competition, rank, isAverage, show, country, key,
```

Comment:
I'm pretty sure this isn't how to use keys. The key should just be provided to `ResultRow`, but not explicitly used in `ResultRow`.

Reply:
Yeah, you _cannot_ access a prop named `key`; it will always be undefined. It's a reserved name, basically.

Reply:
> I fixed that issue by setting <Table.Body key={show}> to force a rerender of the table when the show mode is changed. @kr-matthews if you want to investigate this more feel free, but I spend an hour and that was the only thing that worked

The issue is that you have duplicate keys, specifically the `result.id` `6350611` is coming up multiple times, both when 'show' is 'results' and 'by region'. For the latter, I believe this is because the WR is also the NAR and the "USR", so that key comes up 3 times. I'm not sure why it's coming up twice for 'results'.

Reply:
No it really is a duplicate key (result id and show), the console was printing warnings.

Reply:
oh I get it now. The issue is that the duplicate rows are not across categories, but in the by-region category which react then doesn't clean up I guess.

Reply:
fixed by adding country.name

Reply:
That doesn't fix it when 'show' is 'results'. I'm still seeing the duplicate key warning:
`Warning: Encountered two children with the same key, `6350611-100 results-China`.` This is at `results/rankings/333/single?show=100+results` i.e. load the rankings page and click 'Results'.

Reply:
I have now used tanstack's react table to handle the key generation